s3upload.js

```diff
@@ -15,6 +15,10 @@ S3Upload.prototype.onFinishS3Put = function(signResult, file) {
     return console.log('base.onFinishS3Put()', signResult.publicUrl);
 };
 
+S3Upload.prototype.preprocess = function(file, next) {
+    return console.log('base.preprocess()', file);
+};
```

Comment:
Probably should still call `next(file)` here, even though this isn't really used.  But it would be broken if we ever fell back to this function.